packages/core/src/utils/requestQueue.ts

```diff
@@ -104,10 +125,63 @@ export const createRequestQueue = ({
     }
   };
 
+  const subscribe = async (request: Omit<SubscribeParameters, "method">) => {
+    for (let i = 0; i <= RETRY_COUNT; i++) {
+      try {
+        const stopClock = startClock();
+
+        const wsTransport = resolveWebsocketTransport(network.transport);
+
+        if (wsTransport === undefined) {
+          throw new Error(
```

Comment:
I think this should be `NonRetryableError`

Reply:
agree, it should always be defined by the way we make use of _eth_subscribe in sync-realtime.